packages/components/types/components/Select.d.ts

```diff
@@ -6,6 +6,19 @@ import {
 } from './_shared';
 import { BaseFieldProps } from './BaseField';
 
+type SelectOption = {
+	id?: string;
+	value?: string;
+	disabled?: boolean;
+	label?: string;
+};
+
+type SelectOptionGroup = {
+	id?: string;
+	label?: string;
+	options?: Array<SelectOption>;
```

Comment:
I understand why `SelectOption`'s properties are optional but I think `options` and `label` at least are required, no? Personally I would make `id` required too (for both) even though that slightly augments the requirements for the consumer because it absolutely prevents the consumer from programming an unsafe select.

Reply:
> I think options and label at least are required, no

That's a good point! I'll make that update 🙏 .

> Personally I would make id required too (for both)

That was something I wasn't sure about 🤔 . The default HTML use IDs for `optgroup` and `option` in that way.

I figured I'd provide it as an option for the consumer since data typically flows into Select-like components with that structure (which would be prioritized for the React `key` prop).